common/cards/card-plugins/effects/totem.js

```diff
@@ -9,31 +8,37 @@ class TotemEffectCard extends EffectCard {
 			name: 'Totem',
 			rarity: 'ultra_rare',
 			description:
-				'Player recovers +10hp after being knocked out and remains in battle.\n\nDiscard when applied.',
+				'Recover 10hp and remain in battle after you are knocked out.\nDoes not count as a knockout. Discard after use.',
 		})
-		this.recoverAmount = 10
 	}
 
 	/**
+	 *
 	 * @param {GameModel} game
+	 * @param {string} instance
+	 * @param {import('../../../types/cards').CardPos} pos
 	 */
-	register(game) {
-		// attacks
-		game.hooks.attack.tap(this.id, (target) => {
-			if (target.effectCardId === this.id) {
-				target.recovery.push({amount: this.recoverAmount, discardEffect: true})
-			}
-			return target
-		})
-
+	onAttach(game, instance, pos) {
 		game.hooks.hermitDeath.tap(this.id, (recovery, deathInfo) => {
```

Comment:
Use the afterAttack hook and this needs to be redone to work with that hook, make sure to discard the totem.